skbio/sequence/_sequence.py

```diff
@@ -1242,6 +1240,62 @@ def slices_from_regex(self, regex, ignore=None):
                 yield slice(lookup[match.start(g)],
                             lookup[match.end(g) - 1] + 1)
 
+    def iter_contiguous(self, included, min_length=1, invert=False):
+        """Yield contiguous subsequences based on `included`.
+
+        Parameters
+        ----------
+        included : 1D array_like (bool) or iterable (slices or ints)
+            `included` is transformed into a flat boolean vector where each
+            position will either be included or skipped. All contiguous
+            included positions will be yielded as a single region.
+        min_length : int, optional
+            The minimum length of a subsequence for it to be yielded.
+            Default is 1.
+        invert : bool, optional
+            Whether to invert `included` such that it describes what should be
+            skipped instead of included. Default is False.
+
+        Returns
+        -------
+        generator
+            Yields subsequences as indicated by `included`.
+
+        Notes
+        -----
+        If slices provide adjacent ranges, then they will be considered the
```

Comment:
Ah I missed that -- I think it'd help to either add a description to that example noting the adjacent behavior or add a new example with adjacent slices.